File: 240207/빙빙 돌며 사각형 채우기/snail-alphabet-square.py
n, m = map(int, input().split())
arr = [[0 for _ in range(m)] for _ in range(n)]

def in_range(x, y):
    return 0 <= x < n and 0 <= y < m

dx, dy =[0, 1, 0, -1], [1, 0, -1, 0]
x, y = 0, 0
arr[x][y] = 'A'
dir_num = 0

for i in range(1, n*m):
    nx, ny = x + dx[dir_num], y + dy[dir_num]

    if not in_range(nx, ny) or arr[nx][ny] != 0:
        dir_num = (dir_num + 1) % 4
    
    x, y = x + dx[dir_num], y + dy[dir_num]
    arr[x][y] = chr(ord('A') + i)
    
for i in range(n):
    for j in range(m):
        print(arr[i][j], end=' ')
    print()

File: 240914/빙빙 돌며 사각형 채우기/snail-alphabet-square.js
const fs = require("fs");
let [n, m] = fs.readFileSync(0).toString().trim().split(" ").map(Number);

let arr = Array(n).fill(0).map(() => Array(m).fill(0));
let char = "A"
arr[0][0] = char;

let dx = [0, 1, 0, -1], dy = [1, 0, -1, 0];
let [x, y] = [0, 0];
let dirNum = 0;

for (let i = 1; i < n * m; i++) {
    let nx = x + dx[dirNum], ny = y + dy[dirNum];

    if (!inRange(nx, ny) || arr[nx][ny] !== 0) {
        dirNum = (dirNum + 1) % 4;
    }

    x += dx[dirNum], y += dy[dirNum];
    char = String.fromCharCode(char.charCodeAt(0) + 1);
    if (char > "Z") {
        char = "A";
    }
    arr[x][y] = char;
}

for (let i = 0; i < n; i++) {
    console.log(arr[i].join(" "));
}

function inRange(x, y) {
    return x >= 0 && x < n && y >= 0 && y < m;
}